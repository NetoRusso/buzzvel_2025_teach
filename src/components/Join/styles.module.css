.container {
  width: 100%;
  min-height: 800px;
  padding: 60px 20px;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: var(--yellow-600);
  position: relative;
  overflow: hidden;
}

.content {
  width: 100%;
  max-width: 1500px;
  display: flex;
  align-items: center;
  justify-content: center;
  position: relative;
  z-index: 1;
}

.box {
  width: 100%;
  min-width: 300px;
  height: auto;
  min-height: 320px;
  display: flex;
  flex-flow: column nowrap;
  align-items: center;
  justify-content: center;
  gap: 32px;
  text-align: center;
  position: relative;
  z-index: 2;
}

.title {
  font-size: clamp(1.5rem, 0.7493rem + 3.0564vw, 3.5rem);
  line-height: 1.15;
  font-weight: 800;
  color: var(--blue-Gray-900);
}

.description {
  width: 100%;
  font-size: clamp(1rem, 0.9062rem + 0.382vw, 1.25rem);
  line-height: 1.6;
  font-weight: 400;
  color: var(--blue-Gray-900);
  max-width: 450px;
}

.btn {
  width: clamp(184px, calc(100vw * 239 / 1440), 239px);
  max-width: 239px;
  min-width: 184px;
  height: clamp(56px, calc(100vw * 64 / 1440), 64px);
  max-width: 64px;
  min-height: 56px;
  border-radius: 8px;
  border: none;
  background-color: var(--orange-600);
  font-size: clamp(1.25rem, 1.1562rem + 0.382vw, 1.5rem);
  font-weight: 700;
  color: var(--background-white);
  cursor: pointer;
  transition: all 0.3s ease-in-out;
}

.btn:hover {
  scale: 0.95;
  background-color: transparent;
  color: var(--blue-Gray-900);
  border: 2px solid var(--blue-Gray-900);
}

.visuals_container {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  pointer-events: none;
  z-index: 0;
}

.image_wrapper {
  position: absolute;
  will-change: transform;
}

.image_wrapper img {
  display: block;
  max-width: 100%;
  height: auto;
}

.user_photo_wrapper img {
  border-radius: 50%;
  object-fit: cover;
}

.element_float_wrapper img {
  border-radius: 0;
}