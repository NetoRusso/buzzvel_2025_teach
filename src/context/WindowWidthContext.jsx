'use client'
import { createContext, useState, useEffect } from "react";

const WindowWidthContext = createContext();

export const WindowWidthProvider = ({ children }) => {
  const [windowWidth, setWindowWidth] = useState(0);
  const [isMobile, setIsMobile] = useState(false);

  useEffect(() => {
    const handleResize = () => {
      setWindowWidth(window.innerWidth);
    }

    window.addEventListener("resize", handleResize);

    handleResize();

    return () => {
      window.removeEventListener("resize", handleResize);
    };
  }, []);

  useEffect(() => {
    if(windowWidth < 840) {
      setIsMobile(true);

    } else if(windowWidth > 840) {
      setIsMobile(false);

    }
  }, [windowWidth]);

  return (
    <WindowWidthContext.Provider value={{ windowWidth, isMobile }}>
      {children}
    </WindowWidthContext.Provider>
  );
};

export { WindowWidthContext };