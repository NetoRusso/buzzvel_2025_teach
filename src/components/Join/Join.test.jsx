import { render, screen, act, waitFor } from '@testing-library/react';
import Join from './index';
import { WindowWidthContext } from '@/context/WindowWidthContext';
import { useInView } from 'react-intersection-observer';

// Mocks para assets de imagem (PNG e SVG)
// É importante que os caminhos aqui correspondam exatamente a como são importados no componente Join.jsx
// Se você usa require('./assets/upl01.png').default, o mock deve refletir isso.
// Para simplificar, vamos assumir que o mock de next/image lida com o objeto src.
const mockImageSrc = (name) => ({ default: { src: `/mock/${name}.png`, height: 50, width: 50, blurDataURL: 'data:...' }});
jest.mock('./assets/upl01.png', () => mockImageSrc('upl01'));
jest.mock('./assets/upl02.png', () => mockImageSrc('upl02'));
jest.mock('./assets/upl03.png', () => mockImageSrc('upl03'));
jest.mock('./assets/upl04.png', () => mockImageSrc('upl04'));
jest.mock('./assets/upl05.png', () => mockImageSrc('upl05'));
jest.mock('./assets/upl06.png', () => mockImageSrc('upl06'));
jest.mock('./assets/upl07.png', () => mockImageSrc('upl07'));
jest.mock('./assets/upl08.png', () => mockImageSrc('upl08'));
jest.mock('./assets/upd01.png', () => mockImageSrc('upd01'));
jest.mock('./assets/upd02.png', () => mockImageSrc('upd02'));
jest.mock('./assets/upd03.png', () => mockImageSrc('upd03'));
jest.mock('./assets/upd04.png', () => mockImageSrc('upd04'));
jest.mock('./assets/upd05.png', () => mockImageSrc('upd05'));
jest.mock('./assets/upd06.png', () => mockImageSrc('upd06'));

jest.mock('./assets/il01.svg', () => 'il01.svg');
jest.mock('./assets/il02.svg', () => 'il02.svg');
jest.mock('./assets/il03.svg', () => 'il03.svg');
jest.mock('./assets/il04.svg', () => 'il04.svg');
jest.mock('./assets/il05.svg', () => 'il05.svg');
jest.mock('./assets/il06.svg', () => 'il06.svg');
jest.mock('./assets/id01.svg', () => 'id01.svg');
jest.mock('./assets/id02.svg', () => 'id02.svg');
jest.mock('./assets/id03.svg', () => 'id03.svg');
jest.mock('./assets/id04.svg', () => 'id04.svg');
jest.mock('./assets/id05.svg', () => 'id05.svg');

jest.mock('next/image', () => ({
  __esModule: true,
  default: (props) => {
    const { src, alt, width, height, ...rest } = props;
    const imgSrc = typeof src === 'string' ? src : (src && src.src);
    // eslint-disable-next-line @next/next/no-img-element, jsx-a11y/alt-text
    return <img src={imgSrc || 'mock-src'} alt={alt} width={width} height={height} {...rest} />;
  },
}));

jest.mock('react-intersection-observer', () => ({
  useInView: jest.fn(),
}));

const mockGsapTo = jest.fn();
const mockGsapSet = jest.fn();
const mockScrollTriggerKill = jest.fn();
const mockScrollTriggerGetAll = jest.fn(() => []);
const mockGsapKillTweensOf = jest.fn();
let mockGsapRegistered = false;

jest.mock('gsap', () => {
  const actualGsap = jest.requireActual('gsap');
  return {
    gsap: {
      registerPlugin: (...plugins) => {
        mockGsapRegistered = true;
      },
      to: (...args) => {
        mockGsapTo(...args);
        const stInstance = { kill: mockScrollTriggerKill, trigger: args[0] };
        return { scrollTrigger: stInstance };
      },
      set: mockGsapSet,
      killTweensOf: mockGsapKillTweensOf,
      utils: actualGsap.gsap.utils,
    },
  };
});

jest.mock('gsap/ScrollTrigger', () => ({
  ScrollTrigger: {
    getAll: mockScrollTriggerGetAll,
    kill: mockScrollTriggerKill,
  }
}));

describe('Join Component', () => {
  let mockUseInViewValue;
  const mockSetWindowWidth = jest.fn();

  const renderJoinComponent = (isMobileContext = false, inViewContext = false, customWindowWidth = isMobileContext ? 390 : 1024) => {
    mockUseInViewValue = {
      ref: jest.fn(),
      inView: inViewContext,
      entry: undefined,
    };
    useInView.mockReturnValue(mockUseInViewValue);

    return render(
      <WindowWidthContext.Provider value={{ windowWidth: customWindowWidth, isMobile: isMobileContext, setWindowWidth: mockSetWindowWidth }}>
        <Join />
      </WindowWidthContext.Provider>
    );
  };

  beforeEach(() => {
    mockGsapRegistered = false;
    mockGsapTo.mockClear();
    mockGsapSet.mockClear();
    mockScrollTriggerKill.mockClear();
    mockScrollTriggerGetAll.mockReturnValue([]);
    mockGsapKillTweensOf.mockClear();
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('renders static content correctly', () => {
    renderJoinComponent();
    expect(screen.getByRole('heading', { name: /Join a world of learning/i })).toBeInTheDocument();
    expect(screen.getByText(/Malesuada ut aliquam at ac est nisi/i)).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /Sign Up Now/i })).toBeInTheDocument();
  });

  it('renders all user photos and icons initially', () => {
    renderJoinComponent();
    const userPhotos = screen.getAllByAltText('User');
    expect(userPhotos.length).toBe(14);
    const icons = screen.getAllByAltText('Icon');
    expect(icons.length).toBe(11);
  });

  it('does not initialize GSAP animations if not in view', async () => {
    renderJoinComponent(false, false);
    // Pequena espera para garantir que nenhum import() assíncrono seja disparado
    await act(async () => {
        await new Promise(resolve => setTimeout(resolve, 0));
    });
    expect(mockGsapRegistered).toBe(false);
    expect(mockGsapTo).not.toHaveBeenCalled();
  });

  it('initializes GSAP and sets desktop positions when in view and not mobile', async () => {
    renderJoinComponent(false, true); 

    await waitFor(() => expect(mockGsapRegistered).toBe(true));
    await waitFor(() => expect(mockGsapSet).toHaveBeenCalled());
    
    const upl01Element = screen.getByTestId('visual-item-upl01');
    expect(mockGsapSet).toHaveBeenCalledWith(upl01Element, expect.objectContaining({ top: '-15%', left: '26%', scale: 1 }));
    expect(mockGsapTo).toHaveBeenCalledWith(upl01Element, expect.objectContaining({ yPercent: -35 }));
  });

  it('initializes GSAP and sets mobile positions and animations when in view and mobile', async () => {
    renderJoinComponent(true, true);

    await waitFor(() => expect(mockGsapRegistered).toBe(true));
    await waitFor(() => expect(mockGsapSet).toHaveBeenCalled());

    const upl01Element = screen.getByTestId('visual-item-upl01');
    expect(mockGsapSet).toHaveBeenCalledWith(upl01Element, expect.objectContaining({ top: '-60%', left: '85%', scale: expect.any(Number) }));
    expect(mockGsapTo).toHaveBeenCalledWith(upl01Element, expect.objectContaining({ xPercent: -30 }));
  });
  
  it('cleans up GSAP animations when isMobile or windowWidth changes after GSAP is loaded', async () => {
    const { rerender } = renderJoinComponent(false, true); 

    await waitFor(() => expect(mockGsapTo).toHaveBeenCalled());
    mockGsapTo.mockClear(); 
    mockGsapSet.mockClear();
    mockScrollTriggerGetAll.mockReturnValue([{ trigger: document.createElement('div'), kill: mockScrollTriggerKill }]);

    const mobileContextValue = { windowWidth: 400, isMobile: true, setWindowWidth: mockSetWindowWidth };
    act(() => { 
      rerender(
        <WindowWidthContext.Provider value={mobileContextValue}>
          <Join />
        </WindowWidthContext.Provider>
      );
    });
    
    await waitFor(() => {
      expect(mockScrollTriggerKill).toHaveBeenCalled();
      expect(mockGsapKillTweensOf).toHaveBeenCalled();
    });
    await waitFor(() => expect(mockGsapTo).toHaveBeenCalled());
    if (mockGsapTo.mock.calls.length > 0) { 
        const firstGsapToCallArgs = mockGsapTo.mock.calls[0][1];
        expect(firstGsapToCallArgs).toHaveProperty('xPercent');
    } else {
        throw new Error("gsap.to was not called after re-render for mobile");
    }
  });

  it('matches snapshot on desktop when in view', async () => {
    const { container } = renderJoinComponent(false, true);
    await waitFor(() => expect(screen.getByTestId('visual-item-upl01')).toBeInTheDocument());
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot on mobile when in view', async () => {
    const { container } = renderJoinComponent(true, true);
    await waitFor(() => expect(screen.getByTestId('visual-item-upl01')).toBeInTheDocument());
    expect(container.firstChild).toMatchSnapshot();
  });
});