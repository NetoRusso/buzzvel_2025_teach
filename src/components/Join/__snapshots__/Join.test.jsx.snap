// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Join Component matches snapshot on desktop when in view 1`] = `
<section
  aria-label="Join us"
  class="container "
>
  <div
    class="content"
  >
    <div
      class="box"
    >
      <h2
        class="title"
      >
        Join a world of learning
      </h2>
      <p
        class="description"
      >
        Malesuada ut aliquam at ac est nisi, interdum etiam dignissim. Risus elit et fringilla habitant ut facilisi.
      </p>
      <button
        class="btn"
      >
        Sign Up Now
      </button>
    </div>
    <div
      class="visuals_container"
    >
      <div
        class="image_wrapper user_photo_wrapper"
        data-id="upl01"
        data-testid="visual-item-upl01"
        style="width: 60px; height: 60px;"
      >
        <img
          alt="User"
          height="60"
          loading="lazy"
          src="id05.svg"
          width="60"
        />
      </div>
      <div
        class="image_wrapper user_photo_wrapper"
        data-id="upl02"
        data-testid="visual-item-upl02"
        style="width: 48px; height: 48px;"
      >
        <img
          alt="User"
          height="48"
          loading="lazy"
          src="id05.svg"
          width="48"
        />
      </div>
      <div
        class="image_wrapper user_photo_wrapper"
        data-id="upl03"
        data-testid="visual-item-upl03"
        style="width: 64px; height: 64px;"
      >
        <img
          alt="User"
          height="64"
          loading="lazy"
          src="id05.svg"
          width="64"
        />
      </div>
      <div
        class="image_wrapper user_photo_wrapper"
        data-id="upl04"
        data-testid="visual-item-upl04"
        style="width: 96px; height: 96px;"
      >
        <img
          alt="User"
          height="96"
          loading="lazy"
          src="id05.svg"
          width="96"
        />
      </div>
      <div
        class="image_wrapper user_photo_wrapper"
        data-id="upl05"
        data-testid="visual-item-upl05"
        style="width: 48px; height: 48px;"
      >
        <img
          alt="User"
          height="48"
          loading="lazy"
          src="id05.svg"
          width="48"
        />
      </div>
      <div
        class="image_wrapper user_photo_wrapper"
        data-id="upl06"
        data-testid="visual-item-upl06"
        style="width: 120px; height: 120px;"
      >
        <img
          alt="User"
          height="120"
          loading="lazy"
          src="id05.svg"
          width="120"
        />
      </div>
      <div
        class="image_wrapper user_photo_wrapper"
        data-id="upl07"
        data-testid="visual-item-upl07"
        style="width: 32px; height: 32px;"
      >
        <img
          alt="User"
          height="32"
          loading="lazy"
          src="id05.svg"
          width="32"
        />
      </div>
      <div
        class="image_wrapper user_photo_wrapper"
        data-id="upl08"
        data-testid="visual-item-upl08"
        style="width: 48px; height: 48px;"
      >
        <img
          alt="User"
          height="48"
          loading="lazy"
          src="id05.svg"
          width="48"
        />
      </div>
      <div
        class="image_wrapper user_photo_wrapper"
        data-id="upd01"
        data-testid="visual-item-upd01"
        style="width: 56px; height: 56px;"
      >
        <img
          alt="User"
          height="56"
          loading="lazy"
          src="id05.svg"
          width="56"
        />
      </div>
      <div
        class="image_wrapper user_photo_wrapper"
        data-id="upd02"
        data-testid="visual-item-upd02"
        style="width: 96px; height: 96px;"
      >
        <img
          alt="User"
          height="96"
          loading="lazy"
          src="id05.svg"
          width="96"
        />
      </div>
      <div
        class="image_wrapper user_photo_wrapper"
        data-id="upd03"
        data-testid="visual-item-upd03"
        style="width: 64px; height: 64px;"
      >
        <img
          alt="User"
          height="64"
          loading="lazy"
          src="id05.svg"
          width="64"
        />
      </div>
      <div
        class="image_wrapper user_photo_wrapper"
        data-id="upd04"
        data-testid="visual-item-upd04"
        style="width: 128px; height: 128px;"
      >
        <img
          alt="User"
          height="128"
          loading="lazy"
          src="id05.svg"
          width="128"
        />
      </div>
      <div
        class="image_wrapper user_photo_wrapper"
        data-id="upd05"
        data-testid="visual-item-upd05"
        style="width: 48px; height: 48px;"
      >
        <img
          alt="User"
          height="48"
          loading="lazy"
          src="id05.svg"
          width="48"
        />
      </div>
      <div
        class="image_wrapper user_photo_wrapper"
        data-id="upd06"
        data-testid="visual-item-upd06"
        style="width: 40px; height: 40px;"
      >
        <img
          alt="User"
          height="40"
          loading="lazy"
          src="id05.svg"
          width="40"
        />
      </div>
      <div
        class="image_wrapper element_float_wrapper"
        data-id="il01"
        data-testid="visual-item-il01"
        style="width: 56px; height: 56px;"
      >
        <img
          alt="Icon"
          height="56"
          loading="lazy"
          src="id05.svg"
          width="56"
        />
      </div>
      <div
        class="image_wrapper element_float_wrapper"
        data-id="il02"
        data-testid="visual-item-il02"
        style="width: 96px; height: 96px;"
      >
        <img
          alt="Icon"
          height="96"
          loading="lazy"
          src="id05.svg"
          width="96"
        />
      </div>
      <div
        class="image_wrapper element_float_wrapper"
        data-id="il03"
        data-testid="visual-item-il03"
        style="width: 40px; height: 40px;"
      >
        <img
          alt="Icon"
          height="40"
          loading="lazy"
          src="id05.svg"
          width="40"
        />
      </div>
      <div
        class="image_wrapper element_float_wrapper"
        data-id="il04"
        data-testid="visual-item-il04"
        style="width: 32px; height: 32px;"
      >
        <img
          alt="Icon"
          height="32"
          loading="lazy"
          src="id05.svg"
          width="32"
        />
      </div>
      <div
        class="image_wrapper element_float_wrapper"
        data-id="il05"
        data-testid="visual-item-il05"
        style="width: 48px; height: 48px;"
      >
        <img
          alt="Icon"
          height="48"
          loading="lazy"
          src="id05.svg"
          width="48"
        />
      </div>
      <div
        class="image_wrapper element_float_wrapper"
        data-id="il06"
        data-testid="visual-item-il06"
        style="width: 60px; height: 60px;"
      >
        <img
          alt="Icon"
          height="60"
          loading="lazy"
          src="id05.svg"
          width="60"
        />
      </div>
      <div
        class="image_wrapper element_float_wrapper"
        data-id="id01"
        data-testid="visual-item-id01"
        style="width: 104px; height: 104px;"
      >
        <img
          alt="Icon"
          height="104"
          loading="lazy"
          src="id05.svg"
          width="104"
        />
      </div>
      <div
        class="image_wrapper element_float_wrapper"
        data-id="id02"
        data-testid="visual-item-id02"
        style="width: 64px; height: 64px;"
      >
        <img
          alt="Icon"
          height="64"
          loading="lazy"
          src="id05.svg"
          width="64"
        />
      </div>
      <div
        class="image_wrapper element_float_wrapper"
        data-id="id03"
        data-testid="visual-item-id03"
        style="width: 48px; height: 48px;"
      >
        <img
          alt="Icon"
          height="48"
          loading="lazy"
          src="id05.svg"
          width="48"
        />
      </div>
      <div
        class="image_wrapper element_float_wrapper"
        data-id="id04"
        data-testid="visual-item-id04"
        style="width: 48px; height: 48px;"
      >
        <img
          alt="Icon"
          height="48"
          loading="lazy"
          src="id05.svg"
          width="48"
        />
      </div>
      <div
        class="image_wrapper element_float_wrapper"
        data-id="id05"
        data-testid="visual-item-id05"
        style="width: 32px; height: 32px;"
      >
        <img
          alt="Icon"
          height="32"
          loading="lazy"
          src="id05.svg"
          width="32"
        />
      </div>
    </div>
  </div>
</section>
`;

exports[`Join Component matches snapshot on mobile when in view 1`] = `
<section
  aria-label="Join us"
  class="container mobile_layout"
>
  <div
    class="content"
  >
    <div
      class="box"
    >
      <h2
        class="title"
      >
        Join a world of learning
      </h2>
      <p
        class="description"
      >
        Malesuada ut aliquam at ac est nisi, interdum etiam dignissim. Risus elit et fringilla habitant ut facilisi.
      </p>
      <button
        class="btn"
      >
        Sign Up Now
      </button>
    </div>
    <div
      class="visuals_container"
    >
      <div
        class="image_wrapper user_photo_wrapper"
        data-id="upl01"
        data-testid="visual-item-upl01"
        style="width: 60px; height: 60px;"
      >
        <img
          alt="User"
          height="60"
          loading="lazy"
          src="id05.svg"
          width="60"
        />
      </div>
      <div
        class="image_wrapper user_photo_wrapper"
        data-id="upl02"
        data-testid="visual-item-upl02"
        style="width: 48px; height: 48px;"
      >
        <img
          alt="User"
          height="48"
          loading="lazy"
          src="id05.svg"
          width="48"
        />
      </div>
      <div
        class="image_wrapper user_photo_wrapper"
        data-id="upl03"
        data-testid="visual-item-upl03"
        style="width: 64px; height: 64px;"
      >
        <img
          alt="User"
          height="64"
          loading="lazy"
          src="id05.svg"
          width="64"
        />
      </div>
      <div
        class="image_wrapper user_photo_wrapper"
        data-id="upl04"
        data-testid="visual-item-upl04"
        style="width: 96px; height: 96px;"
      >
        <img
          alt="User"
          height="96"
          loading="lazy"
          src="id05.svg"
          width="96"
        />
      </div>
      <div
        class="image_wrapper user_photo_wrapper"
        data-id="upl05"
        data-testid="visual-item-upl05"
        style="width: 48px; height: 48px;"
      >
        <img
          alt="User"
          height="48"
          loading="lazy"
          src="id05.svg"
          width="48"
        />
      </div>
      <div
        class="image_wrapper user_photo_wrapper"
        data-id="upl06"
        data-testid="visual-item-upl06"
        style="width: 120px; height: 120px;"
      >
        <img
          alt="User"
          height="120"
          loading="lazy"
          src="id05.svg"
          width="120"
        />
      </div>
      <div
        class="image_wrapper user_photo_wrapper"
        data-id="upl07"
        data-testid="visual-item-upl07"
        style="width: 32px; height: 32px;"
      >
        <img
          alt="User"
          height="32"
          loading="lazy"
          src="id05.svg"
          width="32"
        />
      </div>
      <div
        class="image_wrapper user_photo_wrapper"
        data-id="upl08"
        data-testid="visual-item-upl08"
        style="width: 48px; height: 48px;"
      >
        <img
          alt="User"
          height="48"
          loading="lazy"
          src="id05.svg"
          width="48"
        />
      </div>
      <div
        class="image_wrapper user_photo_wrapper"
        data-id="upd01"
        data-testid="visual-item-upd01"
        style="width: 56px; height: 56px;"
      >
        <img
          alt="User"
          height="56"
          loading="lazy"
          src="id05.svg"
          width="56"
        />
      </div>
      <div
        class="image_wrapper user_photo_wrapper"
        data-id="upd02"
        data-testid="visual-item-upd02"
        style="width: 96px; height: 96px;"
      >
        <img
          alt="User"
          height="96"
          loading="lazy"
          src="id05.svg"
          width="96"
        />
      </div>
      <div
        class="image_wrapper user_photo_wrapper"
        data-id="upd03"
        data-testid="visual-item-upd03"
        style="width: 64px; height: 64px;"
      >
        <img
          alt="User"
          height="64"
          loading="lazy"
          src="id05.svg"
          width="64"
        />
      </div>
      <div
        class="image_wrapper user_photo_wrapper"
        data-id="upd04"
        data-testid="visual-item-upd04"
        style="width: 128px; height: 128px;"
      >
        <img
          alt="User"
          height="128"
          loading="lazy"
          src="id05.svg"
          width="128"
        />
      </div>
      <div
        class="image_wrapper user_photo_wrapper"
        data-id="upd05"
        data-testid="visual-item-upd05"
        style="width: 48px; height: 48px;"
      >
        <img
          alt="User"
          height="48"
          loading="lazy"
          src="id05.svg"
          width="48"
        />
      </div>
      <div
        class="image_wrapper user_photo_wrapper"
        data-id="upd06"
        data-testid="visual-item-upd06"
        style="width: 40px; height: 40px;"
      >
        <img
          alt="User"
          height="40"
          loading="lazy"
          src="id05.svg"
          width="40"
        />
      </div>
      <div
        class="image_wrapper element_float_wrapper"
        data-id="il01"
        data-testid="visual-item-il01"
        style="width: 56px; height: 56px;"
      >
        <img
          alt="Icon"
          height="56"
          loading="lazy"
          src="id05.svg"
          width="56"
        />
      </div>
      <div
        class="image_wrapper element_float_wrapper"
        data-id="il02"
        data-testid="visual-item-il02"
        style="width: 96px; height: 96px;"
      >
        <img
          alt="Icon"
          height="96"
          loading="lazy"
          src="id05.svg"
          width="96"
        />
      </div>
      <div
        class="image_wrapper element_float_wrapper"
        data-id="il03"
        data-testid="visual-item-il03"
        style="width: 40px; height: 40px;"
      >
        <img
          alt="Icon"
          height="40"
          loading="lazy"
          src="id05.svg"
          width="40"
        />
      </div>
      <div
        class="image_wrapper element_float_wrapper"
        data-id="il04"
        data-testid="visual-item-il04"
        style="width: 32px; height: 32px;"
      >
        <img
          alt="Icon"
          height="32"
          loading="lazy"
          src="id05.svg"
          width="32"
        />
      </div>
      <div
        class="image_wrapper element_float_wrapper"
        data-id="il05"
        data-testid="visual-item-il05"
        style="width: 48px; height: 48px;"
      >
        <img
          alt="Icon"
          height="48"
          loading="lazy"
          src="id05.svg"
          width="48"
        />
      </div>
      <div
        class="image_wrapper element_float_wrapper"
        data-id="il06"
        data-testid="visual-item-il06"
        style="width: 60px; height: 60px;"
      >
        <img
          alt="Icon"
          height="60"
          loading="lazy"
          src="id05.svg"
          width="60"
        />
      </div>
      <div
        class="image_wrapper element_float_wrapper"
        data-id="id01"
        data-testid="visual-item-id01"
        style="width: 104px; height: 104px;"
      >
        <img
          alt="Icon"
          height="104"
          loading="lazy"
          src="id05.svg"
          width="104"
        />
      </div>
      <div
        class="image_wrapper element_float_wrapper"
        data-id="id02"
        data-testid="visual-item-id02"
        style="width: 64px; height: 64px;"
      >
        <img
          alt="Icon"
          height="64"
          loading="lazy"
          src="id05.svg"
          width="64"
        />
      </div>
      <div
        class="image_wrapper element_float_wrapper"
        data-id="id03"
        data-testid="visual-item-id03"
        style="width: 48px; height: 48px;"
      >
        <img
          alt="Icon"
          height="48"
          loading="lazy"
          src="id05.svg"
          width="48"
        />
      </div>
      <div
        class="image_wrapper element_float_wrapper"
        data-id="id04"
        data-testid="visual-item-id04"
        style="width: 48px; height: 48px;"
      >
        <img
          alt="Icon"
          height="48"
          loading="lazy"
          src="id05.svg"
          width="48"
        />
      </div>
      <div
        class="image_wrapper element_float_wrapper"
        data-id="id05"
        data-testid="visual-item-id05"
        style="width: 32px; height: 32px;"
      >
        <img
          alt="Icon"
          height="32"
          loading="lazy"
          src="id05.svg"
          width="32"
        />
      </div>
    </div>
  </div>
</section>
`;
